src/Installers/Windows/tasks/GenerateGuid.cs

```diff
@@ -13,19 +13,25 @@ public class GenerateGuid : Task
         [Output]
         public string Guid { get; private set; }
 
-        [Required]
         public string NamespaceGuid { get; set; }
 
-        [Required]
         public ITaskItem[] Values { get; set; }
 
         public override bool Execute()
         {
             try
             {
-                var value = string.Join(",", Values.Select(o => o.ItemSpec).ToArray()).ToLowerInvariant();
+                if (!string.IsNullOrEmpty(NamespaceGuid))
+                {
+                    var value = string.Join(",", Values.Select(o => o.ItemSpec).ToArray()).ToLowerInvariant();
 
-                Guid = Uuid.Create(new Guid(NamespaceGuid), value).ToString();
+                    Guid = Uuid.Create(new Guid(NamespaceGuid), value).ToString();
+                }
+                else
+                {
+                    // Providing a guid of * as the product code 
+                    Guid = "*";
```

Comment:
I'm concerned this may cause us to unintentionally break ourselves. I'd rather have a property in the .wixproj file which specifies whether not guids should be generated. So for example, ANCM .wixproj files could set a property `<DisableGuidGeneration>true</DisableGuidGeneration>`. Then, update this target to only run on `Condition=" '$(DisableGuidGeneration)' != 'true' "`
 
https://github.com/aspnet/Universe/blob/da48de7d7580ce316784dd3e250786ffa7017c25/src/Installers/Windows/Wix.targets#L15

Reply:
Disagree with `DisableGuidGeneration`. We are still generating a guid, we just it to be random. Maybe we can add a bool to the GenerateGuid Task called `GenerateRandomGuid`?

Reply:
What's the point of invoking an MSBuild task to set the GUID to `*`?